pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -186,6 +186,11 @@ func (cb *ClusterBuilder) buildSubsetCluster(
 		return nil
 	}
 
+	if len(cb.req.Push.Mesh.OutboundClusterStatName) != 0 {
+		subsetCluster.cluster.AltStatName = telemetry.BuildStatPrefix(cb.req.Push.Mesh.OutboundClusterStatName,
```

Comment:
why set again as it has been set in `buildCluster`

Reply:
We do not have subset name there so SUBSET_NAME wont be replaced

Reply:
We can get the subset from the clustername. 

This way, we do some duplicate work on other fields

Reply:
You have to parse it for every cluster and get subset name - currently this repetition only happens for subset clusters . IMO this is much faster because subset clusters would be few in number.  Unless we change the current signature and pass subset name - we can avoid this.